src/server/routes/api/document/index.js

```diff
@@ -73,6 +79,16 @@ http.post('/', function( req, res ){
     .then( runLayout )
     .then( getDocJson )
     .then( json => res.json( json ) )
+    .catch( e => {
+
+      if( e instanceof Promise.TimeoutError ){
+        res.sendStatus(504);
+      } else {
+        logger.error('Could not fill doc from text: ', `text: ${text}`, e);
```

Comment:
Did you text the output of this?  I'm not sure that Winston logs objects with a prototype like this properly.  Maybe it's best to use a single template string.  You could use `${e.toString()}` to guarantee you get a printable representation.

Reply:
Maybe better to have the error as a separate logger.error() call

Reply:
I agree with the second call to error with just the response, first point needs testing, javascript is all but an exact science.